unstructured_inference/layout_model.py

```diff
@@ -0,0 +1,133 @@
+import os
+import tempfile
+
+import cv2
+import jsons
+import numpy as np
+import onnxruntime
+from huggingface_hub import hf_hub_download
+from pdf2image import convert_from_path
+
+from unstructured_inference.inference.layout import DocumentLayout, LayoutElement, PageLayout
+from unstructured_inference.visualize import vis
+from unstructured_inference.yolox_functions import demo_postprocess, multiclass_nms
+from unstructured_inference.yolox_functions import preproc as preprocess
+
+# NOTE(benjamin) Repository and file to download from hugging_face
+REPO_ID = "unstructuredio/yolo_x_layout"
+FILENAME = "yolox_l0.05.onnx"
+
+LAYOUT_CLASSES = [
+    "Caption",
+    "Footnote",
+    "Formula",
+    "List-item",
+    "Page-footer",
+    "Page-header",
+    "Picture",
+    "Section-header",
+    "Table",
+    "Text",
+    "Title",
+]
+# Path of the model, will be filled in local_inference
+YOLOX_MODEL = ""
+output_dir = "outputs/"
+
+
+def local_inference(filename, type="image", to_json=False, keep_output=False):
```

Comment:
Since this is specific to `yolox`, I'd name this `yolox_local_inference` since this repo will also have code for donut, layoutlm, etc